### PR TITLE
cherry-pick the #12759

### PR DESCRIPTION
**Pull request by wopeizl**
test=develop

### CODE REVIEW

File: paddle/fluid/framework/operator.cc [hunk 1]
```diff
@@ -157,27 +162,59 @@ RuntimeContext::RuntimeContext(const VariableNameMap& innames,
 }
 
 void OperatorBase::Run(const Scope& scope, const platform::Place& place) {
-  VLOG(4) << place << " " << DebugStringEx(&scope);
-  if (platform::is_gpu_place(place)) {
+  try {
+    if (VLOG_IS_ON(4)) {
```
Comment (panyx0718) [inline paddle/fluid/framework/operator.cc:166]: why need VLOG_IS_ON?

File: paddle/fluid/framework/operator.cc [hunk 2]
```diff
@@ -157,27 +162,59 @@ RuntimeContext::RuntimeContext(const VariableNameMap& innames,
...
+    if (platform::IsProfileEnabled()) {
+      platform::DeviceContextPool& pool =
+          platform::DeviceContextPool::Instance();
+      platform::RecordEvent record_event(Type(), pool.Get(place));
+      RunImpl(scope, place);
+    } else {
+      RunImpl(scope, place);
+    }
+
+    if (VLOG_IS_ON(3)) {
```
Comment (panyx0718) [inline paddle/fluid/framework/operator.cc:191]: same as above

**Review by panyx0718 — APPROVED** [review]

File: paddle/fluid/framework/operator.cc (hunk 1 above)
Comment (wopeizl) [reply paddle/fluid/framework/operator.cc:166]: Would romove the cover later.

File: paddle/fluid/framework/operator.cc (hunk 2 above)
Comment (wopeizl) [reply paddle/fluid/framework/operator.cc:191]: Would romove the cover later.

**Pull request merged**